src/app/modules/navigator/test-exec-navigator.component.ts

```diff
@@ -116,25 +122,59 @@ export class TestExecNavigatorComponent implements OnInit, OnDestroy {
       };
       this.messagingService.publish(TEST_EXECUTION_STARTED, payload);
       this.log('sending TEST_EXECUTION_STARTED', payload);
+      await this.testExecutionObserver(response, tclPath);
+      this.switchToIdleStatus();
     } catch (reason) {
-      this.switchToTestRunButton();
+      this.switchToIdleStatus();
       const payload = {
         path: tclPath,
         reason: reason,
         message: 'The test "\${}" could not be started.'
       };
       this.messagingService.publish(TEST_EXECUTION_START_FAILED, payload);
-      this.setupTestSelectedListener();
       this.log('sending TEST_EXECUTION_START_FAILED', payload);
     }
   }
 
-  private switchToTestCancelButton(): void {
+  async testExecutionObserver(testId: string, tclPath: string) {
+    let suiteStatus: TestSuiteExecutionStatus;
+    let executionStatus = TestExecutionState.Running;
+    while (executionStatus === TestExecutionState.Running) {
+      this.log('polling test status from', testId);
+      suiteStatus = await this.testExecutionService.getStatus(testId);
```

Comment:
what if there is a (communication) error? Should we catch the resulting error here, and maybe allow for a couple of retries before giving up?